### ISSUE TITLE
Control Metadata for "Implemented By" and "Assurance

### ISSUE THREAD

**Issue by aj-stein-nist** [usnistgov/oscal-content#97]
# User Story:

As an OSCAL {stakeholder}, I {provide a clear and concise description of what the problem is. Ex. I need to be able to do}

## Goals:

{A clear and concise description of what you want to happen. This should be outcome focused. Include concise description of any alternative solutions or features you've considered. Feel free to include screenshots or examples about the feature request here.}

## Dependencies:

{Describe any previous issues or related work that must be completed to start or complete this issue.}

## Acceptance Criteria

- [ ] All readme documentation affected by the changes in this issue have been updated.
- [ ] A Pull Request (PR) is submitted that fully addresses the goals of this User Story. This issue is referenced in the PR.
- [ ] The CI-CD build process runs without any reported errors on the PR. This can be confirmed by reviewing that all checks have passed in the PR.

{The items above are general acceptance criteria for all User Stories. Please describe anything else that must be completed for this issue to be considered resolved.}

**Comment by aj-stein-nist** [comment]
Work to add these properties and update the pipeline. Assign to Wendell, AJ wants to help. :-)

**Comment by GaryGapinski** [comment]
The "Implemented By" value of "o/s" seems too weak to adequately describe hybrid controls at the control statement (implementation description) level.

**Comment by aj-stein-nist** [comment]
> The "Implemented By" value of "o/s" seems too weak to adequately describe hybrid controls at the control statement (implementation description) level.

I agree with the sentiment, that being said that might be feedback to discuss more generally with the FISMA Implementation Team, and we have to start somewhere. :-)

**Comment by aj-stein-nist** [comment]
@wendellpiez, should we sync up this week and discuss this work and how we will collaborate on it? Let me know!

**Comment by david-waltermire-nist** [comment]
We can use two properties on a control. `implemented-by` with values `organization` or `system` and `assurance` with maybe `true`/`false` values.

**Comment by aj-stein-nist** [comment]
@wendellpiez is working on a transform pipeline enhancement (in the Gitlab OSCAL internal pipeline tooling) that does the following:

- Takes the official Word (DOCX) copy from the FISMA Implementation Team transforming it into HTML
- A second transform in the pipeline reads out table data to find the "Implemented By" and "Assurance" fields for each control family with its own table
- Take the table data from the intermediate format and adds the necessary props `implementation-level` and `assurance` on each control and sub-control accordingly.

Wendell has sent me draft copies and we met earlier in the morning. Next steps for me to move us along and eventually commit a PR.

- [ ] Data quality check: AJ will compare the current PDF outputs to the resulting catalog (since the source Word file is about 1 year out of date and not the official source info anymore)
- [ ] Work with Wendell to stage a PR targeting the source catalog update and pipeline here
- [ ] Determine how best to manually or programmatically updated the top-level UUID for the source catalog since this may be a significant change. It seems these are typically done manually in the past.

**Comment by wendellpiez** [comment]
Added logic to update top-level UUID along with timestamp, with some whitespace cleanup. The process currently runs with no detected discrepancies.

**Comment by aj-stein-nist** [comment]
> Added logic to update top-level UUID along with timestamp, with some whitespace cleanup. The process currently runs with no detected discrepancies.

OK I guess I check that one off the list then! haha

**Comment by wendellpiez** [comment]
I've also added an XSLT to convert a catalog into an HTML showing a pile of Table C tables (one per family). With some additional CSS it can be made to look just like the PDF.

**Comment by aj-stein-nist** [comment]
> I've also added an XSLT to convert a catalog into an HTML showing a pile of Table C tables (one per family). With some additional CSS it can be made to look just like the PDF.

Oh I missed that bit, thank you very much. That will be useful to speed data quality checks mentioned above. :-)

**Comment by aj-stein-nist** [comment]
OK, now that I figure out issues with my local dev environment, I am back to data quality checks and trying to get the PR ready. Wendell and I have an open question about if and how we will relocate any or some of the transforms into a public location, or just make the content patch. Hopefully, once this PR is more stable, we can do so later next week.

**Comment by aj-stein-nist** [comment]
I did a check of the source PDF/HTML data from the pipeline and things are looking very good and checked those off from the task list above. Will talk with Dave and Wendell about the pipeline code and what to do with it in the context of the PR.

Once I fix the whitespace/newline issue in discussed in https://github.com/usnistgov/oscal-content/pull/124#discussion_r949636415, I can remove the WIP/draft label and submit this for final review by Dave, team, and the larger community.
